### PR TITLE
Refactor manifests.md and registries.md into file-specific references.

### PR DESCRIPTION
**Pull request by ras0219-msft**
See a preview at https://review.learn.microsoft.com/en-us/vcpkg/users/vcpkg-json?branch=dev%2Froschuma%2Fmain

I've heavily rewritten these files, so I recommend instead reading them 'cover to cover' as though they were new documents:

- https://review.learn.microsoft.com/vcpkg/users/vcpkg-configuration-json?branch=dev%2Froschuma%2Fmain
- https://review.learn.microsoft.com/vcpkg/users/vcpkg-json?branch=dev%2Froschuma%2Fmain
- https://review.learn.microsoft.com/vcpkg/users/manifests?branch=dev%2Froschuma%2Fmain
- https://review.learn.microsoft.com/vcpkg/users/registries?branch=dev%2Froschuma%2Fmain

### CODE REVIEW

File: vcpkg/users/examples/versioning.getting-started.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ If you want to upgrade your dependencies, you can bump the minimum version const
 { "builtin-baseline": "3426db05b996481ca31e95fff3734cf23e0f51bc" }
 ```
 
-This field declares the versioning baseline for all ports. Setting a baseline is required to enable versioning, otherwise you will get the current versions on the ports directory. You can run 'git rev-parse HEAD' to get the current commit of vcpkg and set it as the builtin-baseline. See the [`builtin-baseline` documentation](../manifests.md#builtin-baseline) for more information.
+This field declares the versioning baseline for all ports. Setting a baseline is required to enable versioning, otherwise you will get the current versions on the ports directory. You can run 'git rev-parse HEAD' to get the current commit of vcpkg and set it as the builtin-baseline. See the [`"builtin-baseline"` documentation](../vcpkg-json.md#builtin-baseline) for more information.
```
Comment (AugP) [inline vcpkg/users/examples/versioning.getting-started.md:160]: Minor correction: "otherwise you will get the current versions **in** the ports directory". 

Might be worth italicizing "ports" in "ports directory" as well since it's the actual name of that directory.

File: vcpkg/users/vcpkg-configuration-json.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    }
+  ],
+  "overlay-ports": [
+    "./team-ports",
+    "./custom-ports"
+   ],
+  "overlay-triplets": [ "./my-triplets" ]
+}
+```
+This example adds a private registry, `https://github.com/northwindtraders/vcpkg-registry`, as the source for the libraries `beicode` and `beison`. All other ports are found from an internal mirror of the Curated Catalog hosted at `https://internal/mirror/of/github.com/Microsoft/vcpkg`.
```
Comment (AugP) [inline vcpkg/users/vcpkg-configuration-json.md:40]: We need to review if we need to make changes to this example repo. Microsoft has a list of pre-approved fictional names, and I can't remember if "Northwind" was on that list. We also need to check if the logo for that used in that repo is an issue. I don't think this review needs to block this PR though, just something to keep in mind for later.

File: vcpkg/users/vcpkg-json.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+description: Reference documentation for the vcpkg.json file format.
+ms.date: 11/30/2022
+---
+# vcpkg.json Reference
+
+For an overview of using manifests with vcpkg, see [Manifest mode](manifests.md).
+
+Manifests are strict [JSON](https://www.json.org) documents. They cannot contain C++-style comments (`//`) nor trailing commas. However you can use field names that start with `$` to write your comments in any object that has a well-defined set of keys. These comment fields are not allowed in any objects which permit user-defined keys (such as `"features"`).
+
+The latest JSON Schema is available at [vcpkg.schema.json](https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json). IDEs with JSON Schema support such as Visual Studio and Visual Studio Code can use this file to provide IntelliSense and syntax checking. For most IDEs, you should set `"$schema"` in your `vcpkg.json` to this URL (like the above example).
```
Comment (AugP) [inline vcpkg/users/vcpkg-json.md:12]: Suggestion: Do not use the word "IntelliSense" here since that is only applicable to VS & VS Code. Describe what is meant by IntelliSense here. Syntax checking is one thing, autocompletion and syntax colorization could be additional features.

File: vcpkg/users/versioning.md [hunk 4]
```diff
@@ -5,8 +5,7 @@ ms.date: 11/30/2022
 ---
 # Versioning reference
 
-Versioning allows you to deterministically control the precise revisions of dependencies used by
-your project from within your manifest file. Versioning only applies to [Manifest mode](manifests.md).
+Versioning allows you to deterministically control the precise revisions of dependencies used by your project from within your manifest file. Versioning only applies to [Manifest mode](manifests.md).
```
Comment (AugP) [inline vcpkg/users/versioning.md:8]: Suggest rewording last sentence to: "Versioning is only available to Manifest mode users".

**Review by AugP — APPROVED** [review]
Left some minor comments, overall looks very good!

**Comment by ras0219-msft** [comment]
I've moved this to come from a fork instead; see #16
